crates/rpc/rpc/src/debug.rs

```diff
@@ -111,14 +111,7 @@ where
                                 tx_hash: Some(tx_hash),
                                 tx_index: Some(index),
                             }),
-                        )
-                        .map_err(|err| {
-                            results.push(TraceResult::Error {
-                                error: err.to_string(),
-                                tx_hash: Some(tx_hash),
-                            });
-                            err
-                        })?;
+                        )?;
```

Comment:
I wasn't too sure what the goal was here, either to store them and continue or to propagate the error. The code seems to suggest that it should just propagate, as if you'd want to continue the loop, you'd have to provide a default value for state_changes

Reply:
but might be wrong, this was added here https://github.com/paradigmxyz/reth/pull/5743/files, so maybe goal was to accumulate everything?

Reply:
https://github.com/ethereum/go-ethereum/blob/767b00b0b514771a663f3362dd0310fc28d40c25/eth/tracers/api.go#L652-L661

this is the only case where the result error is set in geth which has to do with parallel execution

replaying blocks also results in a simple error and not recorded results.

https://github.com/ethereum/go-ethereum/blob/767b00b0b514771a663f3362dd0310fc28d40c25/eth/tracers/api.go#L613-L616

maybe for traceCallMany a better error would be appropriate but since this Pr only removes redundant code this is okay and we can think about the appropriate behaviour seperately